<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="user_control_topic_name" default="/marslite_navigation/joy_ts"/>

  <!-- Launch the `teleop_joystick` node with `user_direct_control:=false`. -->
  <include file="$(find marslite_navigation)/launch/teleop_joystick.launch">
    <arg name="user_direct_control" value="false"/>
    <arg name="message_enabled" value="false"/>
  </include>

  <!-- Launch the `artificial_potential_field` node -->
  <node pkg="marslite_navigation" type="artificial_potential_field" name="artificial_potential_field" output="screen">
    <param name="user_control_topic_name" value="$(arg user_control_topic_name)"/>
    <rosparam file="$(find marslite_navigation)/config/apf_params.yaml"/>
  </node>
  
</launch>